keps/draft-20180426-security-conformance.md

```diff
@@ -0,0 +1,532 @@
+- Non-security related
+- Details about Kubernetes conformance test
+
+## Proposal
+
+### Security Conformance Profiles
+
+The behavioral/non-behavioral criteria are categorized into a few security conformance profiles for different levels of security requirements:
+
+- `Essential`: (aka `Trusted`) the baseline security requirements for the cluster,
+focusing on the threaten from outside the cluster (via APIs) and
+non-privileged (non-root) access on the host running master and kubelet,
+not covering the threaten from user workloads (non-critical, non-system Pods) running inside the cluster.
+The major use case is operating a secure Kubernetes cluster
+with a group of trusted users whose workloads are trusted (e.g. an internal development team).
+The users will have almost no limit on the features/capabilities provided by the cluster.
+The criteria in this category must be satisfied for a Kubernetes cluster to be secure minimally;
+- `Restricted`: (aka `Untrusted`) in addition to `Essential`,
+it covers the threaten from user workloads to the cluster,
+not among user workloads (e.g. single-tenancy model).
+The major use case is operating a secure Kubernetes cluster for untrusted workloads,
+without taking tenancy-isolation into consideration -
+a compromised/malicious workload may compromise other workloads.
+Certain features/capabilities (e.g. privileged containers) from the cluster won't be available to workloads;
+- `Isolated`: (aka `Multi-tenancy`) in addition to `Restricted`,
+it covers the threaten among user workloads (e.g. multi-tenancy model).
+The major use case is operating a secure Kubernetes cluster to be shared by multiple
+tenants whose workloads are untrusted and isolated.
+A compromised/malicious workload should not be able to compromise workloads belong to other tenants.
+Certain features/capabilities (e.g. privileged containers) from the cluster won't be available to workloads.
+- `Sandboxed`: mostly similar to `Isolated`,
+but the containers are running `sandboxed` using better isolation technology,
+like HyperVisor, gVisor etc.
+The `sandboxed` isolation provides additional layer of protection from attack with
+the capability to offer less-constraint features/capabilities from the cluster,
+e.g. privileged containers are supported.
+The major use case is for sharing the cluster with best isolation and/or with less feature/capability limitation.
+
+Note: a workload is possible to be compromised if it's not properly
+designed/implemented according to security standards, or it's vulnerable.
+This is out-of-scope of this proposal.
+
+The criteria described below is based on a namespace with default settings.
+Like a newly created namespace, with only service account `default`.
+Namespaces with special/privileged settings, like `kube-system`,
+or special namespaces required by some system components
+(e.g. nginx ingress controller) are not covered.
+
+### Behavioral Criteria
+
+#### Config File Permissions
```

Comment:
Seems like this belongs under non-behavioral?

Reply:
It can be behavioral, like we can clearly define the behavior of threat if file permissions are not properly set. I will update the description to focus on behaviors.